packages/visualizations/src/components/utils/ColorsLegend.svelte

```diff
@@ -1,13 +1,63 @@
 <svelte:options immutable={true} />
 
 <script lang="ts">
+    import { onDestroy } from 'svelte';
+    import { debounce } from 'lodash';
     import type { DataBounds, ColorsScale, LegendVariant } from '../types';
     import { defaultCompactLegendNumberFormat } from './formatter';
+
     // options to customize the component
     export let dataBounds: DataBounds;
     export let colorsScale: ColorsScale;
     export let variant: LegendVariant;
     export let title: string;
+
+    // the part below is related to labels rotation
+    let legendWidth: number;
+    let colorBoxWidth: number;
+    const labelsWidth: number[] = [];
+    const labelsHeight: number[] = [];
+    let maxLabelsSize: number;
+    let numberOfLabels: number;
+    let displayVertical: boolean | undefined;
+    let isFirstRotation: boolean = true;
+
+    const handleLabelRotation = (
+        legendW: number,
+        labelW: number[],
+        labelH: number[],
+        colorsScl: ColorsScale
+    ): void => {
+        const isPaletteLegend = colorsScl.type === 'palette';
+        const isDomReady = labelH.length !== 0 && labelW.length !== 0 && legendW;
+        const checkForRotation = isPaletteLegend && isDomReady;
+
+        if (checkForRotation) {
+            isFirstRotation = false;
+            numberOfLabels = colorsScl.colors.length + 1;
+            const availableWidthPerLabel: number = legendW / numberOfLabels - 3;
+            colorBoxWidth = legendW / numberOfLabels;
+            maxLabelsSize = displayVertical ? Math.max(...labelH) : Math.max(...labelW);
+            displayVertical = availableWidthPerLabel < maxLabelsSize;
+        }
+    };
+
+    const rotationDebounce = debounce(handleLabelRotation, 200);
+    const handleRotationLifecycle = (
+        legendW: number,
+        labelW: number[],
+        labelH: number[],
+        colorsScl: ColorsScale
+    ): void => {
+        if (isFirstRotation) {
+            handleLabelRotation(legendW, labelW, labelH, colorsScl);
+        } else {
+            rotationDebounce(legendW, labelW, labelH, colorsScl);
+        }
+    };
+
+    $: handleRotationLifecycle(legendWidth, labelsWidth, labelsHeight, colorsScale);
```

Comment:
```suggestion

    $: if (labelsWidth > 0) {
        rotationDebounce(legendWidth, labelsWidth, labelsHeight, colorsScale);
      }
```
The problem we want to avoid here is calling rotationDebounce on the empty array and then wait before we have values. Instead of exposing a reactive state at component level—which is what I'm targeting with all those request 🙏 —, we just don't trigger unnecessary checks and thus debounce.

Reply:
Indeed ! { leading: true } seems to do the job ! I've made the change and I let you test it again 👍 Thanks for the tip !